stackrox/stackrox
# ROX-12316: Remove deprecated ClusterCVE resource

## Description

The `ClusterCVE` resource (for permission sets) was deprecated in 3.72.
Following the branching of 3.73, the associated code can be removed for 3.74.

## Checklist
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI run

Manual testing:
- Deploy central.
- Go to the `Access Control` section.
- Open the `Permission sets` tab.
- Open one permission set (e.g. Admin), and check that `ClusterCVE` is not in the Resource list anymore.
- Back to the `Permission sets` tab, click the `Create permission set` button, check that the form does not expose `ClusterCVE` in the list of resources (only occurrences of the name should be in the warning banner, and as comment for the `Cluster` resource).

Migration testing:
- Deploy an older central version (e.g. 3.72)
- Go to the `Access Control` section
- Perform the steps listed in manual testing, but checking for the presence of the `ClusterCVE` resource rather than its absence.
- Create a permission set
- With the CLI, ensure that the `ClusterCVE` permission is present in the created permission set.
- Upgrade Central to a newer version (e.g. the one built with this pull request), check for migration steps.
- Repeat the UI manual test step, checking this time for `ClusterCVE` absence.
- Use the CLI to ensure that the `ClusterCVE` permission is not present in the existing permission sets.
